.github/workflows/build.yml

```diff
@@ -27,14 +27,26 @@ jobs:
         run: git config --global --add safe.directory /__w/terraform-cdk/terraform-cdk
       - name: ensure correct user
         run: chown -R root /__w/terraform-cdk
-      - name: Get yarn cache directory path
+      # Setup caches for yarn, and go
+      - name: Get cache directory paths
         id: global-cache-dir-path
-        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+        run: |
+          echo "yarn=$(yarn cache dir)" >> $GITHUB_OUTPUT
+          mkdir -p /usr/local/share/.cache/go
+          echo "go=/usr/local/share/.cache/go" >> $GITHUB_OUTPUT
+      - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
+        with:
+          path: ${{ steps.global-cache-dir-path.outputs.yarn }}
+          key: yarn-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}-build
+          restore-keys: |
+            yarn-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}-
+            yarn-${{ runner.os }}-
```

Comment:
Not a huge fan of this one as it causes false negatives. If we don't have a matching yarn.lock file, let's not use a cache.

Reply:
What do you mean by false negatives? If e.g. one node dependency was changed, all other packages in the cache should still be valid, shouldn't they? From my point of view, it's still better to have e.g. 90% of the packages in the restored cache than to start net new. Am I missing something?

Reply:
Hmm, I didn't think of it that way. Yes, yarn should deal with the changes and not use the wrong dependency just because the cache has it. Nevermind then, ignore this comment.

Reply:
One reason I could see, is a cache containing old dependencies and thus growing over time 🤔😄
But if that happens, we can remove these restore-keys :)